okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -56,7 +56,9 @@ class RealWebSocket(
   private val originalRequest: Request,
   internal val listener: WebSocketListener,
   private val random: Random,
-  private val pingIntervalMillis: Long
+  private val pingIntervalMillis: Long,
+  compression: Boolean = false,
+  contextTakeover: Boolean = false
```

Comment:
First issue here. 
While we enable compression by default, this does not work well for tests, as they have lots of golden values relying on the compression being off. For now, compression options are in OFF state on `RealWebSocket` creation, but are **enabled** by default in the `RealWebSocket.connect` method (which is only used by "client" mode).

This way, in "client" mode, RealWebSocket has compression enabled by default, but in "server" mode compression is disabled by default. Looks like "server" mode is not exposed in the API and is for tests only, so I think it's acceptable.

@swankjesse please share your thoughts/suggestions.

Reply:
@swankjesse , is there anything else I can do here?